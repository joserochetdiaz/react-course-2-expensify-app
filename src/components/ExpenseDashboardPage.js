import React from 'React';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';


const ExpenseDashboardPage = () => (
    <div>
        <ExpenseListFilters />
        <ExpenseList />
    </div>
)
export default ExpenseDashboardPage;