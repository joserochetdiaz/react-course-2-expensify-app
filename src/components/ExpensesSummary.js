import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = (props) => {
    const summary = selectExpensesTotal(props.expenses);
    const expenseWord = props.expenses.length === 1 ? 'expense' : 'expenses';
    return (
        <div>
            
            {props.expenses.length > 0 && <h3>You are now viewing {props.expenses.length} {expenseWord} totalling {numeral(summary).format('$0,0.00')}</h3>}
        </div>
      
    );
};
const mapStateToProps = (state) => ({
    expenses: selectExpenses(state.expenses, state.filters)
});
export default connect(mapStateToProps)(ExpensesSummary);