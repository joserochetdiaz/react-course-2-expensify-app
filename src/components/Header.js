import React from 'react';
import {NavLink} from 'react-router-dom';

const Header = () => (
    <header>
        <h1>Expensify</h1>
        <NavLink to="/" activeClassName="isActive" exact={true}>Home</NavLink>
        <NavLink to="/create" activeClassName="isActive">Create expense</NavLink>
        <NavLink to="/help">Help</NavLink>
    </header>
);

export default Header;