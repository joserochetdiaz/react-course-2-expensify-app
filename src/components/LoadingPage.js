import React from 'react';

const LoadingPage = () => (
    <div>
        <img src="../public/images/loader.gif"/>
    </div>
);

export default LoadingPage;