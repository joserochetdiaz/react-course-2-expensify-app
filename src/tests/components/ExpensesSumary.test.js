import React from 'react';
import {shallow} from 'enzyme';
import {ExpensesSummary} from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';
let wrapper;

beforeEach(() => {
    wrapper = shallow(<ExpensesSummary expenses={[expenses[0], expenses[1]]}/>);
});

test('Should render ExpensesSummary', () => {
    expect(wrapper).toMatchSnapshot();
});