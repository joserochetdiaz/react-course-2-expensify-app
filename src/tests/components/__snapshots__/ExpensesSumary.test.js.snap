// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ExpensesSummary 1`] = `
<div>
  <h3>
    You are now viewing 
    2
     expenses totalling 
    $109,695.00
  </h3>
</div>
`;
